ingest/annotations.py

```diff
@@ -73,38 +72,40 @@ def merge_df(self, first_df, second_df):
 
     def preprocess(self):
         """Ensures that:
-            - Labels are treated as strings
-            - Numeric columns are rounded to 3 decimals points
-            - Group annotations are strings
             - 'NAME' in first header row is capitalized
             - 'TYPE' in second header row is capitalized
         """
-        # Grab column names and convert to strings
-        headers = [str(header) for header in self.file.columns.get_level_values(0)]
-        annot_types = self.file.columns.get_level_values(1)
-        # Lowercase second level. Example: NUMeric -> numeric
-        self.file.rename(
-            columns=lambda col_name: col_name.lower(), level=1, inplace=True
-        )
-        name = list(headers)[0]
-        type = list(annot_types)[0].lower()
+
         # Uppercase NAME and TYPE
-        self.file.rename(columns={name: name.upper(), type: type.upper()}, inplace=True)
-        # Make sure group annotations are treated as strings
-        # only run this assignment if group annotations are present
-        if 'group' in annot_types:
-            group_columns = self.file.xs(
-                "group", axis=1, level=1, drop_level=False
-            ).columns.tolist()
-            self.file[group_columns] = self.file[group_columns].astype(str)
-        # Find numeric columns and round to 3 decimals places and are floats
-        # only run this assignment if numeric annotations are present
-        if 'numeric' in list(annot_types):
+        self.headers[0] = self.headers[0].upper()
+        self.annot_types[0] = self.annot_types[0].upper()
+        self.createDataFrame()
+
+    def createDataFrame(self):
+        """
+        - Create dataframe with proper dtypes to ensure:
+            - Labels are treated as strings (objects)
+            - Numeric annotations are treated as float32
+            - Numeric columns are rounded to 3 decimals points
+        """
+        columns = []
+        dtypes = {}
+        dtypes[self.headers[0]] = 'object'
+        dtypes[self.annot_types[0]] = 'object'
+        for annotation, annot_type in zip(self.headers, self.annot_types):
+            dtypes[annotation] = 'object' if annotation is 'group' else 'float32'
```

Comment:
I've had floats on my mind likely, so I wonder: would [`float16`](https://pandas.pydata.org/pandas-docs/stable/user_guide/gotchas.html#why-not-make-numpy-like-r) suffice here?  If not, are we sure `float64` isn't needed?

Using smaller datatypes until we know we need bigger ones seems like would help constrain memory.

While memory usage is an issue with expression ingest, it's not a known problem with metadata ingest, so this comment is merely something to consider -- definitely not a blocker.

Reply:
I originally put float32. I think it might. However, I'm more worried about precision